docs/source/daml/intro/5_Restrictions.rst

```diff
@@ -66,34 +66,38 @@ For example, the simple Iou in :ref:`simple_iou` allowed the no-op where the ``o
   :start-after: -- TRANSFER_TEST_BEGIN
   :end-before: -- TRANSFER_TEST_END
 
-Similarly, you can write a ``Redeem`` choice, which allows the ``owner`` to redeem an ``Iou`` during business hours on weekdays. The choice doesn't do anything other than archiving the ``SimpleIou``. (This assumes that actual cash changes hands off-ledger:)
+Similarly, you can write a ``Redeem`` choice, which allows the ``owner`` to redeem an ``Iou`` *during business hours on weekdays*. Notice the ``Redeem`` choice implementation below confirms that ``getTime`` returns a value that is during business hours on weekdays. If all those checks pass, it doesn't do anything other than archive the ``SimpleIou``. (This assumes that actual cash changes hands off-ledger:)
 
 .. literalinclude:: daml/daml-intro-5/daml/Restrictions.daml
   :language: daml
   :start-after: -- REDEEM_CHOICE_BEGIN
   :end-before: -- REDEEM_CHOICE_END
 
+There's quite a lot going on inside the ``do`` block of the ``Redeem`` choice, with several uses of the ``<-`` operator. ``do`` blocks and ``<-`` deserve a proper explanation which we will get to below. For now, let's look at how the ``Redeem`` choice is exercised in a script.
+
 .. literalinclude:: daml/daml-intro-5/daml/Restrictions.daml
   :language: daml
   :start-after: -- REDEEM_TEST_BEGIN
   :end-before: -- REDEEM_TEST_END
 
-There are quite a few new time-related functions from the ``DA.Time`` and ``DA.Date`` libraries here. Their names should be reasonably descriptive so how they work won't be covered here, but given that Daml assumes it is run in a distributed setting, we will still discuss time in Daml.
+In the above example, the time is taken apart into day of week and hour of day using standard library functions from ``DA.Date`` and ``DA.Time``. The hour of the day is checked to be in the range from 8 to 18.
 
-There's also quite a lot going on inside the ``do`` block of the ``Redeem`` choice, with several uses of the ``<-`` operator. ``do`` blocks and ``<-`` deserve a proper explanation at this point.
+There are quite a few new time-related functions from the ``DA.Time`` and ``DA.Date`` libraries here. Their names should be reasonably descriptive so how they work won't be covered here, but given that Daml assumes it is run in a distributed setting, we will still discuss time in Daml.
```

Comment:
This made more sense when it was referring to both the choice and the test. Now that you have moved intervening text between the two code snippets, the test only contains two time-related functions and neither of them are from DA.Time or DA.Date (they are in fact from DA.Script). So this text probably needs to be moved up to comment on the choice.

More over I find this style "so how they work won't be covered...." simultaneously condescending and belittling—particularly discouraging if they the reader does not find them "reasonably descriptive". I think it would be better to just provide a reference to their documentation.

Reply:
Fixed with 5daca107a2c08103d21bd3ae93905b776f91efaf